ambry-replication/src/main/java/com/github/ambry/replication/ReplicationEngine.java

```diff
@@ -370,6 +368,17 @@ private int getReplicaThreadIndexToUse(String datacenter) {
     return replicaThreads;
   }
 
+  /**
+   * Chooses a name for a new replica thread.
+   * @param datacenterToReplicateFrom The datacenter that we replicate from in this thread.
+   * @param threadIndexWithinPool The index of the thread within the thread pool.
+   * @return The name of the thread.
+   */
+  protected String getReplicaThreadName(String datacenterToReplicateFrom, int threadIndexWithinPool) {
```

Comment:
If the only difference is the prefix, the method could be getReplicaThreadPrefix() with no arguments.

Reply:
Yep, agreed. Though I named it like this on purpose to
1) prevent the creation of very specialized functions
2) allow the subclass needs to add more details to the thread name if needed
What do you think? I'm also okay with changing it to your suggestion.

Reply:
This is a good change. Thanks @parasgithub for making this change. I don't disagree with your argument to allow the subclasses to add more details to the thread name if needed. 
However, I have a nitpick in that case. I think other than prefix, the rest of the logic is repeated code. Can you separate them out, so that it is ensured that thread naming convention doesn't change inadvertently in subclasses, unless explicitly desired.

Reply:
Done, please take a look